pkg/apicast/apicast.go

```diff
@@ -0,0 +1,357 @@
+package apicast
+
+import (
+	"net/url"
+	"strconv"
+	"strings"
+
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	extensions "k8s.io/api/extensions/v1beta1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+type APIcast struct {
```

Comment:
The idea with this own type, is to have an "internal" type independent of the CRD type so they are independent to changes.

Reply:
Yes, It allows you to convert, filter process the CR information to something else. Besides you can get information not only from CR, but also from ConfigMaps and Secrets. Set them in the internal struct and then use it to build desired k8s objects.